packages/components/src/entities/SamplesDeriveButton.tsx

```diff
@@ -22,57 +23,42 @@ export interface SamplesDeriveButtonProps
 
 export const SamplesDeriveButton: FC<SamplesDeriveButtonProps> = memo(props => {
     const { model, asSubMenu, ...createSampleMenuProps } = props;
-    const [selectionsAreSet, setSelectionsAreSet] = useState<boolean>(false);
-    const [selectionData, setSelectionData] = useState<Map<any, any>>();
-    const selectedCount = useMemo(() => model?.selections?.size ?? -1, [model?.selections]);
-    const useSelectionData = useMemo(() => model?.filterArray.length > 0 && selectedCount <= MAX_EDITABLE_GRID_ROWS , [model?.filterArray]);
+    const { filterArray, isLoadingSelections, schemaQuery, selections } = model;
+    const [selectionData, setSelectionData] = useState<Record<any, any>>();
+    const selectedCount = useMemo<number>(() => selections?.size ?? -1, [selections]);
+    const requestColumns = useMemo<string>(() => {
+        if (!model.queryInfo) return undefined;
+        return model.getRequestColumnsString();
+    }, [model]);
+    const useSelectionData = filterArray.length > 0 && selectedCount > 0 && selectedCount <= MAX_EDITABLE_GRID_ROWS;
```

Comment:
I'm curious why you chose to remove the `useMemo` here. The `filterArray` changes less frequently than selections certainly so perhaps not worth memoizing?  Did you actually notice a difference?

Reply:
I generally use memoization to prevent reprocessing of potentially expensive or uncontrolled logic (from the perspective of this component). When it comes to boolean checks that contain no uncontrolled logic I tend to not optimize for memoization since the memoization itself is another set of boolean checks.